keps/sig-storage/1855-csi-driver-service-account-token/README.md

```diff
@@ -0,0 +1,242 @@
+
+## Table of Contents
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [User stories](#user-stories)
+- [Proposal](#proposal)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [API Changes](#api-changes)
+  - [Example Workflow](#example-workflow)
+  - [Notes/Constraints/Caveats](#notesconstraintscaveats)
+  - [Test Plan](#test-plan)
+  - [Graduation Criteria](#graduation-criteria)
+    - [Alpha-&gt;Beta](#alpha-beta)
+    - [Beta-&gt;GA](#beta-ga)
+- [Production Readiness Review Questionnaire](#production-readiness-review-questionnaire)
+  - [Feature Enablement and Rollback](#feature-enablement-and-rollback)
+  - [Scalability](#scalability)
+- [Alternatives](#alternatives)
+- [Implementation History](#implementation-history)
+<!-- /toc -->
+
+## Summary
+
+This KEP proposes a way to obtain service account token for pods that the CSI
+drivers are mounting volumes for. Since these tokens are valid only for a
+limited period, this KEP will also give the CSI drivers an option to re-execute
+`NodePublishVolume` to mount volumes.
+
+## Motivation
+
+Currently, the only way that CSI drivers acquire service account tokens is to
+directly read the token in the file system. However, this approach has
+uncharming traits:
+
+1.  It will not work for csi drivers which run as a different non-root user than
+    the pods. See
+    [file permission section for service account token](https://github.com/kubernetes/enhancements/blob/master/keps/sig-storage/20180515-svcacct-token-volumes.md#file-permission).
+2.  CSI driver will have access to the secrets of pods that do not use it
+    because the CSI driver should have a `hostPath` volume for the `pods`
+    subdirectory to read the token.
+3.  The audience of the token is defaulted to kube apiserver.
+4.  The token is not guaranteed to be available (e.g.
+    `automountServiceAccountToken=false`).
+
+### User stories
+
+- HashiCorp Vault provider for secret store CSI driver requires service
+  account token of the pods they are mounting secrets at to authenticate to
+  Vaults. The provisioned secrets also have given TTL in Vault, so it is
+  necessary get tokens after the initial mount.
+- Cert manager CSI dirver will create CertificateRequests on behalf of the
+  pods.
+- Amazon EFS CSI driver wants the service account tokens of pods to exchange
+  for AWS credentials.
+
+## Proposal
+
+### Goals
+
+- Allow CSI driver to request audience-bounded service account tokens of pods
+  from kubelet to `NodePublishVolume`.
+- Provide an option to re-execute `NodePublishVolume` in a best-effort manner.
+
+### Non-Goals
+
+- Other CSI calls may not acquire pods' service account tokens via this
+  feature.
+- Failed re-execution of `NodePublishVolume` will not unmount volumes.
+
+### API Changes
+
+```go
+// CSIDriverSpec is the specification of a CSIDriver.
+type CSIDriverSpec struct {
+    ... // existing fields
+
+    RequiresRemount *bool
+    ServiceAccountToken *ServiceAccountToken
```

Comment:
Is there ever a usecase for multiple service account tokens? If I'm using the secret driver with both AWS and valut?

Reply:
done